### ISSUE TITLE
Resolve the new BXXX that appeared when switching to ruff

### ISSUE THREAD

**Issue by hillairet** [Delgan/loguru#968]
As discussed in [this PR](https://github.com/Delgan/loguru/pull/955#issuecomment-1694694943) a few linting and validation errors appeared when we switched from flake8 to ruff.
To get the PR merged, global exclusions of the errors were added to the ruff configuration.
Instead we should deal with each error one by one by fixing the issue or adding a `noqa` wherever it's justified to do so.
For example some (all?) B018 complains about `1 / 0` being a useless expression. But it's part of a test checking that the exception is handled correctly so `noqa` is justified.

**Comment by hillairet** [comment]
@Delgan Question about the errors in `loguru/__init__.pyi`.
Why did you use `__self` rather than `self` for some methods?
You ignored this error in `_logger.py` so I'll ignore line by line in `loguru/__init__.pyi` as well. The question is because I'm curious.  :wink:

**Comment by Delgan** [comment]
Hi @hillairet. :)

I use `__self` in `loguru/__init__.pyi` to match the signature of the `logger` methods defined in `_logger.py`.  These class methods are implemented using `__self` instead of the typical `self` to mitigate the risk of conflicts arising from arbitrary keys employed in `**kwargs` arguments.

In other words, it makes it possible to do the following:

```python
self = "delgan"
logger.info("I'm {self}", self=self)
```

Otherwise, this would raise an exception:

```
TypeError: Logger.info() got multiple values for argument 'self'
```

This also applies to the `__message` parameter.

Ideally, we could use [PEP 570 – Python Positional-Only Parameters](https://peps.python.org/pep-0570/) which is designed for such cases. However, this is only available for Python 3.8, and Loguru supports older versions of Python that are not compatible with this new syntax. Therefore, I prefixed the parameters with "__" as it's a convention used by some libraries to signal their are positional-only.
